{{cookiecutter.project_name}}/.github/workflows/release.yaml

```
@@ -5,26 +5,23 @@ on:
     tags:
       - "*.*.*"
 
+# Use "trusted publishing", see https://docs.pypi.org/trusted-publishers/
 jobs:
   release:
-    name: Release
+    name: Upload release to PyPI
     runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/{{ cookiecutter.package_name }}
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
-
-      - name: Set up Python 3.10
-        uses: actions/setup-python@v4
-        with:
-          python-version: "3.10"
-
-      - name: Install hatch
-        run: pip install hatch
-
-      - name: Build project for distribution
-        run: hatch build
-
-      - name: Publish a Python distribution to PyPI
+     - uses: actions/checkout@v3
+     - uses: actions/setup-python@v4
+       with:
+          python-version: "3.11"
```

```suggestion
          python-version: "3.x"
```

only needs build deps, which shouldn’t be as sensitive as runtime deps like numba